airrohr-firmware/platformio.ini

```diff
@@ -13,42 +13,89 @@ src_dir = .
+  -DVTABLES_IN_FLASH
+build_flags_esp32_release = ${common.build_flags_esp32} -DNDEBUG
+build_flags_esp32_debug = ${common.build_flags_esp32} -g -Og -fno-inline -DJTAG_DEBUGGER=1 -DDEBUG_ESP_PORT=Serial
+
 board_build.f_cpu = 160000000L
-; Always depend on specific library versions (wherever possible)
-; Keep Library names in alphabetical order
+; Always depend on specific library versions (wherever possible and only for external libraries)
+; Keep Library names in the order of their dependencies (leaves at the top)
 ; ESP8266WiFi in both cases is necessary to solve a case sensitivity issue with WiFiUdp.h
-; TinyGPSPlus is not yet in platformio - https://github.com/platformio/platformio-libmirror/issues/99
-lib_deps_external =
+; Use library ID numbers instead of names for libraries whose names are not unique
+;   (like OneWire, LiquidCrystal_I2C and TinyGPSPlus)
+
+lib_deps_generic_external =
+  1@2.3.4 ; OneWire
+  576@1.1.4 ; LiquidCrystal_I2C
+  ;136 ; LiquidCrystal, alternative to 576 LiquidCrystal_I2C?
   Adafruit Unified Sensor@1.0.3
   Adafruit BMP085 Library@1.0.1
   Adafruit BMP280 Library@1.0.4
   Adafruit BME280 Library@1.0.9
   Adafruit HTU21DF Library@1.0.2
-  ArduinoJson@6.11.5
+  ArduinoJson@6.12.0
   DallasTemperature@3.8.0
+  ESP8266_SSD1306@4.0.0
+  1655@1.0.2 ; TinyGPSPlus, formerly this was referenced as mikalhart/TinyGPSPlus#v0.95
+
+lib_deps_esp8266_external =
+  ;ESPSoftwareSerial@3.4.1 ; the latest version available in platformio
+  ;plerup/espsoftwareserial#5.2.9 ; this version needs unpublished 8266 plaform, see https://github.com/plerup/espsoftwareserial/issues/103
+  plerup/espsoftwareserial#5.0.3 ; this version compiles with a standard 8266 platform
+
+lib_deps_esp32_external =
+
+; system libraries from platform -> no version number
+lib_deps_esp8266_platform =
+  Hash
+  Wire
+  SPI
+  ESP8266WiFi
   DNSServer
   ESP8266HTTPClient
   ESP8266WebServer
-  ESP8266WiFi
-  ESP8266_SSD1306@4.0.0
   ESP8266httpUpdate
   ESP8266mDNS
-  OneWire@2.3.4
-  SPI
-  mikalhart/TinyGPSPlus#v0.95
+
+; system libraries from platform -> no version number
+lib_deps_esp32_platform =
   Wire
-  LiquidCrystal_I2C@1.1.2
-  ESPSoftwareSerial@3.4.1
+  SPI
+  WiFi
+  DNSServer
+  WiFiClientSecure
+  HTTPClient
+  FS
+  SPIFFS
+  WebServer
+  Update
+  HTTPUpdate
+  ESPmDNS
+
+lib_deps_esp8266 = ${common.lib_deps_esp8266_platform} ${common.lib_deps_esp8266_external} ${common.lib_deps_generic_external}
+lib_deps_esp32 = ${common.lib_deps_esp32_platform} ${common.lib_deps_esp32_external} ${common.lib_deps_generic_external}
+
 extra_scripts = platformio_script.py
 # This release is reflecting the Arduino Core 2.4.2 release
 # When the requirement for Arduino Core is raised, this
 # needs to be adjusted to the matching version from
 # https://github.com/platformio/platform-espressif8266/releases
-platform_version = espressif8266@1.8.0
+;platform_version = espressif8266@1.8.0
+platform_version = espressif8266@2.2.3 ; using Arduino core 2.5.2
```

Comment:
please update the comment above - but we can't upgrade to newer arduino core until OTA problem is solved.

Reply:
already reverted. I have to switch back and forth all the time...

I'm not sure about using which SoftwareSerial variant and version.

Reply:
What OTA problem?

Reply:
firmware size>512kB cannot be flashed using OTA:
4 MB Flash 3MB SPIFFS, 512 kB running firmware, 512kB new firmware

Reply:
by the way: what is the real use of mDNS? Removing it would free some flash and the old version is pretty buggy

Reply:
I'm working on a two-stage OTA path that will allow to OTA firmwares >= 512kb. but it is halted by all the other problems in the code base, I can not well test my implementation right now, so I'm working on the other problems first. 

I believe the usecase for mdns is to export the $hostname.local as MDNS entry so that you can e.g. access http://esp8266-foobar.local/ and get to the IP address.. this is interesting in case of us providing https at some point (where certificates on hostnames are much nicer than on IP addresses). 

I personally would also like to get rid of the DNSServer dependency, it is fairly large and only needed for the initial setup (in STA_AP mode).

Reply:
but if we complain about size, I would just postpone the addition of the SPS30 sensor until we have solved the size problems.. this way we could relase OTAs with all the bugfixes and move forward.

Reply:
to use "http://esp8266-<ID>.local/" you need to know the <ID> (part of the MAC address) before you connect to it. Some kind of catch22.
That's why I think it is not really useful.

Reply:
> I would just postpone the addition of the SPS30 sensor

Good idea.

I also have a implemented a new sensor (MHZ19 CO2 sensor) but I didn't dare to create a pull-request, yet. Maybe I will do it in the next days just to store it somewhere.

Reply:
I often use it to quickly see the current PM value of my own sensor.

mDNS assigns a name to the node, but also adverttises the services running on the node (e.g. webserver, even the Arduino OTA port if you have local OTA enabled). On Android you can see this with https://play.google.com/store/apps/details?id=com.druk.servicebrowser for example, on linux you can see local services with avahi-discover,

Reply:
@nospam2000 the name can be discovered in the initial setup step where you connect to it in ap mode. Also I thought you can configure it..

Reply:
mDNS is used to discover available dust sensors with the firmware flashing tool (https://github.com/opendata-stuttgart/airrohr-firmware-flasher). So usersdon't need to know how to look up the IP address (this project is also for users without much IT knowlegde). 
DNSServer is needed for the "login" feature while setting up the sensor. Most OSes send a DNS request for special host names to test internet connectivity. These requests need to be answered.

Reply:
The description of the firmware flasher says _"connect the ESP8266 with a USB cable to your computer (micro-USB)"_. Can it really also be used for OTA flashing as a **local** OTA update? Actually that is what I was looking for because the USB upload is a pain.

Reply:
Not at the moment. Maybe we can implement something similar with the update version @dirkmueller is working on (load new firmware to SPIFFS, update actual firmware to a small binary that will then only copy the new version from SPIFFS to flash)